snuba/datasets/entities/sessions.py

```diff
@@ -211,8 +212,24 @@ def __init__(self) -> None:
     def select_storage(
         self, query: Query, request_settings: RequestSettings
     ) -> StorageAndMappers:
-        granularity = extract_granularity_from_query(query, "started") or 3600
-        use_materialized_storage = granularity >= 3600 and (granularity % 3600) == 0
+        use_materialized_storage: Optional[bool] = None
+
+        # If the passed in `request_settings` arg is an instance of `SubscriptionRequestSettings`,
+        # then it is a crash rate alert subscription, and hence we decide on whether to use the
+        # materialized storage or the raw storage by examining the time_window.
+        # If the `time_window` <=1h, then select the raw storage otherwise select materialized
+        # storage
+        # NOTE: If we were to support other types of subscriptions over the sessions dataset that
+        # do not follow this method used to identify which storage to use, we would need to
+        # find a different way to distinguish them.
+        if isinstance(request_settings, SubscriptionRequestSettings):
+            from_date, to_date = get_time_range(query, "started")
+            if from_date and to_date:
```

Comment:
No, because we will just fall to the logic we already have which will set granularity to 3600 i.e. materialised storage

Reply:
It looks like it would only fall to materialised storage if `extract_granularity_from_query(query, "started")` is None or a multiple of 3600. What value are we expecting to get there for subscriptions?

Reply:
It should never actually not have `from_date` and `to_date` in the subscription queries, I just added the check as a safe guard but if it does fail for whatever reason and fallsback to the old logic then `extract_granularity_from_query(query, "started")` will always return `None` because we don't group by any column (time column in this case) and there is validation against any grouping logic actually for subscription queries.

Reply:
Ok, in that case I think you can rewrite in a way that makes it clearer that the fallback is always materialised and we don't attempt to look for non existent "started" value. Like:
```
        if isinstance(request_settings, SubscriptionRequestSettings):
            from_date, to_date = get_time_range(query, "started")
            if from_date and to_date:
                use_materialized_storage = to_date - from_date > timedelta(hours=1)
            else:
                use_materialized_storage = True
        else:
            granularity = extract_granularity_from_query(query, "started") or 3600
            use_materialized_storage = granularity >= 3600 and (granularity % 3600) == 0
```

However, I would question whether we should be falling back at all. If there is no `from_date` or `to_date` wouldn't we be better off raising an invalid subscription rather than attempting to run a broken query. I think it is better to always reject a incorrectly formatted subscription than try to make it valid which may cause us problems down the line.

Reply:
>However, I would question whether we should be falling back at all. If there is no from_date or to_date wouldn't we be better off raising an invalid subscription/query error rather than attempting to run a broken query

I don't agree here because there is a required `time_window` param that is passed to Subscription classes, and from that the `from_date`, and `to_date` are generated so if we get to this point, we will always have `from_date`, and `to_date`. However, there is something to be said for the readability of this code segment and not having to go through the extracting the granularity from time columns that we know don't exist in a subscription query, so I will use your snippet